### DISCUSSION TITLE
Question about key file length

### DISCUSSION THREAD

**Discussion by th-sommer** [keepassxreboot/keepassxc#9904] [Q&A] [answered]
I'm trying to understand how a key file enhances the master password. I found a discussion in which droidmonkey described the calculation of the master key as  `sha256(seed + sha256(sha256(password) + sha256(filedata)))`.
I admit, it's all greek to me.
So my question is:
Would a random 2048bit key file be "better" than the standard 128bit file generated by KeePassXC.

**Comment by droidmonkey** [comment, marked as answer]
No it would not be better because it is being reduced to 256 bits of key material anyway. I believe the new key files are 256 bit now. 

Edit: key file version 2 is a sha256 hash of 128 bits of random data.
